miner/algo_common.go

```diff
@@ -181,21 +185,40 @@ func applyTransactionWithBlacklist(signer types.Signer, config *params.ChainConf
 }
 
 // commit tx to envDiff
-func (envDiff *environmentDiff) commitTx(tx *types.Transaction, chData chainData) (*types.Receipt, int, error) {
-	header := envDiff.header
-	coinbase := &envDiff.baseEnvironment.coinbase
-	signer := envDiff.baseEnvironment.signer
+func (envDiff *environmentDiff) commitTx(tx *types.Transaction, chData chainData, algoConf algorithmConfig) (*types.Receipt, int, error) {
+	var (
+		header   = envDiff.header
+		coinbase = &envDiff.baseEnvironment.coinbase
+		signer   = envDiff.baseEnvironment.signer
+		root     = envDiff.state.OriginalRoot()
+	)
 
 	gasPrice, err := tx.EffectiveGasTip(header.BaseFee)
 	if err != nil {
 		return nil, shiftTx, err
 	}
 
 	envDiff.state.SetTxContext(tx.Hash(), envDiff.baseEnvironment.tcount+len(envDiff.newTxs))
-
 	receipt, newState, err := applyTransactionWithBlacklist(signer, chData.chainConfig, chData.chain, coinbase,
 		envDiff.gasPool, envDiff.state, header, tx, &header.GasUsed, *chData.chain.GetVMConfig(), chData.blacklist)
 
+	// when drop transaction is enabled:
+	// 1. if there was an error applying the transaction OR
+	// 2. if the transaction receipt status is failed
+	// we don't apply the transaction to the state and we don't add it to the newTxs, return early
+	if algoConf.DropTransactionOnRevert && (err != nil || (receipt != nil && receipt.Status == types.ReceiptStatusFailed)) {
+		// the StateDB for environment diff is already modified at this point, since it gets mutated when passed in to
+		// applyTransactionWithBlacklist, so we need to revert it
+		s, sdbErr := state.New(root, envDiff.state.Database(), chData.chain.Snapshots())
```

Comment:
We only exclude when flag is enabled and transaction is in reverting transaction hashes list, logic can be found in `commitBundle`

```
		// if drop transaction on revert is enabled and the transaction is found in list of reverting transaction hashes,
		// we can consider the transaction for discarding
		canDiscard = discard && bundle.OriginalBundle.RevertingHash(tx.Hash())
		algoConf.DropTransactionOnRevert = canDiscard

```

Reply:
> why would we want to exclude reverting transactions though

This was the feature request - product requirement was that:
```
Update builder logic such that when reverting transaction hashes are specified, the builder discards transactions on revert for bundles and sbundles
```

Reply:
Perhaps I misunderstood. Updated the logic based on @Ruteri and @dvush feedback. 

> if it can't be included (e.g. nonce error) - we just skip it (new thing)

This is now what is implemented in this PR. Please un-resolve if there is still issue.